types/eslint-utils/index.d.ts

```diff
@@ -0,0 +1,217 @@
+ */
+export function isParenthesized(times: number, node: Node, sourceCode: SourceCode): boolean;
+/**
+ * Check whether a given node is parenthesized or not.
+ * @param node The AST node to check.
+ * @param sourceCode The source code object to get tokens.
+ * @returns `true` if the node is parenthesized.
+ */
+export function isParenthesized(node: Node, sourceCode: SourceCode): boolean;
+
+/**
+ * Checks if the given token is an arrow token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is an arrow token.
+ */
+export function isArrowToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a comma token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a comma token.
+ */
+export function isCommaToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a semicolon token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a semicolon token.
+ */
+export function isSemicolonToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a colon token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a colon token.
+ */
+export function isColonToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is an opening parenthesis token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is an opening parenthesis token.
+ */
+export function isOpeningParenToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a closing parenthesis token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a closing parenthesis token.
+ */
+export function isClosingParenToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is an opening square bracket token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is an opening square bracket token.
+ */
+export function isOpeningBracketToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a closing square bracket token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a closing square bracket token.
+ */
+export function isClosingBracketToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is an opening brace token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is an opening brace token.
+ */
+export function isOpeningBraceToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a closing brace token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a closing brace token.
+ */
+export function isClosingBraceToken(token: AST.Token): boolean;
+
+/**
+ * Checks if the given token is a comment token or not.
+ * @param token The token to check.
+ * @returns `true` if the token is a comment token.
+ */
+export function isCommentToken(token: AST.Token): boolean;
```

Comment:
We need to be able to pass Comment as an argument.

```suggestion
export function isCommentToken(token: AST.Token | import('estree').Comment):  token is import('estree').Comment;
```

Reply:
As can be seen in https://github.com/typescript-eslint/typescript-eslint/pull/3421, there's no need to do `AST.Token | import('estree').Comment`

https://github.com/typescript-eslint/typescript-eslint/blob/e5a030d511507e3fe79848561ddd6e066c7f01e1/packages/experimental-utils/src/ast-utils/eslint-utils/predicates.ts#L64-L66

Reply:
`eslint.AST.Token` does not contain any Comments.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/eslint/index.d.ts#L18

Reply:
Is this a bug that should be fixed in `@types/eslint` then?

Reply:
I'm not the maintainer of eslint so I'm not sure, but I think it could be a `@types/eslint` bug, as you say 😅

Reply:
Err, I've lost the thread. What is the requested change?

Reply:
`eslint.AST.Token` should include `estree.Comment`

Reply:
I'll add a `type CommentOrToken = Comment | AST.Token;` here, since `Comment` comes from `estree` and `token` comes from `eslint`. 👍